os/pm/pm_activity.c

```diff
@@ -202,6 +202,28 @@ void pm_stay(int domain, enum pm_state_e state)
 	leave_critical_section(flags);
 }
 
+/****************************************************************************
+ * Name: pm_set_timer
+ *
+ * Description:
+ *   This function is called to set a timed callback to an intended function.
+ *   It may be used to invoke pm_relax() after a fixed time duration of
+ *   locking using pm_stay().
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   None.
+ *
+ ****************************************************************************/
+
+void pm_set_timer(void)
```

Comment:
I would like to know your opinion on calling a function like `up_set_pm_timer()` which is in `amebasmart_pmhelpers.c`, directly from `os/pm/pm_*.c` files

We felt that it might be better to have another layer in between so we chose this design. Would like to know your ideas on this please.

Also, what would be your opinion about using a different name like `pm_timer_wrapper()` instead of `pm_set_timer()`?

Reply:
pm function which manages a timer for pm operation can have a name with pm_xxxx.
But the `up_set_timer` can be a general timer API, I don't think that BSP should have pm specific timer API. Because it does not need to do something for pm only.
1. pm timer function should manage all timer usages for uart, ble and so on.
2. pm timer function should call timer BSP API or hw timer API with open, read, write as @edwakuwaku said.

Reply:
Modified the control flow as follows: -
1. APP tries to lock/unlock file with timer functionality
2. pm_procfs.c redirects timer setting call to pm_set_timer()
3. pm_set_timer() sets the timer attributes and calls BSP layer helper function for PM
4. BSP layer PM helper function calls unified timer setting API for AI Dual

**_Steps 3 and 4 may be clubbed, PENDING_**

Reply:
@sunghan-chang requesting your review here please